frontend/packages/dev-console/src/components/import/git/SourceSecretSelector.tsx

```diff
@@ -1,46 +1,91 @@
+/* eslint-disable no-unused-vars, no-undef */
 import * as React from 'react';
-import { FormGroup, ControlLabel, FormControl, HelpBlock } from 'patternfly-react';
+import { FormGroup, ControlLabel } from 'patternfly-react';
+import { useFormikContext, FormikValues, useField } from 'formik';
+import { createModalLauncher } from '../../../../../../public/components/factory/modal';
 import SourceSecretDropdown from '../../dropdown/SourceSecretDropdown';
-import CreateSourceSecret, { SecretType } from './CreateSourceSecret';
+import {
+  withSecretForm,
+  SourceSecretForm,
+  SecretTypeAbstraction,
+} from '../../../../../../public/components/secrets/create-secret';
 
 const CREATE_SOURCE_SECRET = 'create-source-secret';
 
 interface SourceSecretSelectorProps {
-  namespace?: string;
-  sourceSecret: string;
-  selectedKey: string;
-  secretCredentials: { [key: string]: string };
-  onChange: (
-    name: string,
-    key: string,
-    authType?: string,
-    credentials?: { [key: string]: string },
-  ) => void;
+  namespace: string;
 }
 
-const SourceSecretSelector: React.FC<SourceSecretSelectorProps> = ({
-  sourceSecret,
+interface CreateSourceSecretModalProps {
+  cancel: (e: Event) => void;
+  close: () => void;
+  onSave?: (name: string) => void;
+  onCancel: () => void;
+  namespace: string;
+}
+
+const CreateSourceSecretModal: React.FC<CreateSourceSecretModalProps> = ({
+  close,
   namespace,
-  secretCredentials,
-  selectedKey,
-  onChange,
+  onSave,
+  onCancel,
 }) => {
-  const onDropdownChange = (secretName: string, key: string) => {
-    if (key === CREATE_SOURCE_SECRET) {
-      onChange('', key, SecretType.basicAuth, {});
-    } else {
-      onChange(secretName, key);
-    }
+  const CreateSourceSecretForm = withSecretForm(SourceSecretForm, true);
+  const onClose = () => {
+    close();
+    onCancel();
+    return null;
   };
 
-  const onInputChange: React.ReactEventHandler<HTMLInputElement> = (event) => {
-    onChange(event.currentTarget.value, selectedKey, SecretType.basicAuth, secretCredentials);
+  return (
+    <CreateSourceSecretForm
+      onCancel={onClose}
+      onSave={onSave}
+      fixed={{ metadata: { namespace } }}
+      secretTypeAbstraction={SecretTypeAbstraction.source}
+      explanation="Source secrets let you authenticate against a Git server."
+      titleVerb="Create"
+      isCreate
+    />
+  );
+};
+
+const sourceSecretModalLauncher = createModalLauncher<CreateSourceSecretModalProps>(
+  CreateSourceSecretModal,
+);
+
+const SourceSecretSelector: React.FC<SourceSecretSelectorProps> = ({ namespace }) => {
+  const [selectedKey] = useField('git.secret.selectedKey');
+  const { setFieldValue } = useFormikContext<FormikValues>();
+  let secretName: string;
```

Comment:
This isn't right. The secret name needs too be derived from the form state. It cannot be a local variable.